From: =?UTF-8?q?Rog=C3=A9rio=20Brito?= <rbrito@ime.usp.br>
Date: Thu, 24 Oct 2013 01:11:21 -0200
Subject: Provide command line option -a

Create a new command line option (-a) for the fsck.hfsplus that has the same
behavior that the -p option has, for greater compatibility with other tools.
---
 fsck_hfs.tproj/fsck_hfs.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/fsck_hfs.tproj/fsck_hfs.c
+++ b/fsck_hfs.tproj/fsck_hfs.c
@@ -104,7 +104,7 @@ main(argc, argv)
 	else
 		progname = *argv;
 
-	while ((ch = getopt(argc, argv, "c:D:dfglm:npqruy")) != EOF) {
+	while ((ch = getopt(argc, argv, "c:D:dfglm:napqruy")) != EOF) {
 		switch (ch) {
 		case 'c':
 			/* Cache size to use in fsck_hfs */
@@ -169,6 +169,7 @@ main(argc, argv)
 			yflag = 0;
 			break;
 
+		case 'a':
 		case 'p':
 			preen++;
 			break;
@@ -572,7 +573,7 @@ usage()
 	(void) fprintf(stderr, "  l = live fsck (lock down and test-only)\n");
 	(void) fprintf(stderr, "  m arg = octal mode used when creating lost+found directory \n");
 	(void) fprintf(stderr, "  n = assume a no response \n");
-	(void) fprintf(stderr, "  p = just fix normal inconsistencies \n");
+	(void) fprintf(stderr, "  p, a = just fix normal inconsistencies \n");
 	(void) fprintf(stderr, "  q = quick check returns clean, dirty, or failure \n");
 	(void) fprintf(stderr, "  r = rebuild catalog btree \n");
 	(void) fprintf(stderr, "  u = usage \n");
